src/audio/dai-zephyr.c

```diff
@@ -1461,6 +1462,7 @@ int dai_common_copy(struct dai_data *dd, struct comp_dev *dev, pcm_converter_fun
 		 dev->direction, copy_bytes);
 
 	/* Check possibility of glitch occurrence */
+#ifndef CONFIG_ZEPHYR_DP_SCHEDULER
```

Comment:
Replied in #8621 , where this patch got split out.  Please continue discussion there.  I'll remove it from this PR in the next spin as we can just cherry pick during development.

Reply:
> I'll remove it from this PR in the next spin as we can just cherry pick during development.

This sounds like you stick to one pull request per branch and one branch per pull request which may explain why this big PR initially mixed separate features.

It's actually very easy to submit multiple pull requests from a single branch. Just rotate the commits in (ma)git with an interactive rebase before pushing. I almost never submit HEAD to a pull request, always HEAD~n.

No need to constantly cherry-pick commits from different branches when there is only one.

- Single pull request with distinct features: bad.
- Single pull git branch with distinct features: great!

In case someone else missed this link the first time:
https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs